src/gradualizer_prelude.erl

```diff
@@ -18,3 +19,81 @@
       Par :: atom(),
       Def :: term(),
       Val :: any().
+
+-spec lists:append([[T]])     -> [T].
+-spec lists:append([T], [T])  -> [T].
+-spec lists:delete(T, [T])    -> [T].
+-spec lists:droplast([T, ...]) -> [T].
+
+-spec lists:dropwhile(fun((T) -> boolean()), [T]) -> [T].
+
+-spec lists:duplicate(T, non_neg_integer()) -> [T].
+
+-spec lists:filter(fun((T) -> boolean() | {true, Value}), [T]) -> [T | Value].
+
+-spec lists:flatmap(fun((A) -> B), [A]) -> [B].
+
+-type deep_list(A) :: [A | deep_list(A)].
+
+-spec lists:flatten(deep_list(A))      -> [A].
+-spec lists:flatten(deep_list(A), [A]) -> [A].
+
+-spec lists:foldl(fun((T, Acc) -> Acc), [T], Acc) -> Acc.
+-spec lists:foldr(fun((T, Acc) -> Acc), [T], Acc) -> Acc.
+-spec lists:join(T, [T]) -> [T].
+
+-spec lists:foreach(fun((T) -> term()), [T]) -> ok.
+
+-spec lists:last([T, ...]) -> T.
+
+-spec lists:map(fun((A) -> B), [A]) -> [B].
+-spec lists:mapfoldl(fun((A, Acc) -> {B, Acc}), Acc, [A]) -> {[B], Acc}.
+-spec lists:mapfoldr(fun((A, Acc) -> {B, Acc}), Acc, [A]) -> {[B], Acc}.
+
+-spec lists:max([T, ...]) -> T.
+-spec lists:min([T, ...]) -> T.
+
+-spec lists:merge([[T]]) -> [T].
+-spec lists:merge([X], [Y]) -> [X | Y].
```

Comment:
Why not
```suggestion
-spec lists:merge([T], [T]) -> [T].
```
like `append`?

Reply:
I'm in two minds about this one. I agree that your suggested type is simpler and in some ways better. But the type I gave is similar to that of `merge/3` and there is something to be said about them having similar types for symmetry.